src/runtime_src/core/common/xrt_xclbin.cpp

```diff
@@ -74,3 +78,348 @@ xrtXclbinUUID(xclDeviceHandle dhdl, xuid_t out)
+}
+
+// Utility function for device class to verify that the C xclbin handle is valid
+// Needed when the C API for device tries to load an xclbin using C pointer to xclbin
+namespace xrt_core {
+namespace xclbin_int {
+void
+is_valid_or_error(xrtXclbinHandle handle)
+{
+  if ((xclbins.find(handle) == xclbins.end()))
+    throw xrt_core::error(-EINVAL, "Invalid xclbin handle");
+}
+
+const std::vector<char>&
+get_xclbin_data(xrtXclbinHandle handle)
+{
+  return get_xclbin(handle)->get_data();
+}
+
+}
+};
+
+xrtXclbinHandle
+xrtXclbinAllocFilename(const char* filename)
+{
+  try {
+    auto xclbin = std::make_shared<xrt::xclbin_impl>(filename);
+    auto handle = xclbin.get();
+    xclbins.emplace(handle, std::move(xclbin));
+    return handle;
+  }
+  catch (const xrt_core::error& ex) {
+    xrt_core::send_exception_message(ex.what());
+    errno = ex.get();
+  }
+  catch (const std::exception& ex) {
+    send_exception_message(ex.what());
+    errno = 0;
+  }
+  return nullptr;
+}
+
+xrtXclbinHandle
+xrtXclbinAllocRawData(const char* data, const int size)
+{
+  try {
+    std::vector<char> raw_data(data, data + size);
+    auto xclbin = std::make_shared<xrt::xclbin_impl>(raw_data);
+    auto handle = xclbin.get();
+    xclbins.emplace(handle, std::move(xclbin));
+    return handle;
+  }
+  catch (const xrt_core::error& ex) {
+    xrt_core::send_exception_message(ex.what());
+    errno = ex.get();
+  }
+  catch (const std::exception& ex) {
+    send_exception_message(ex.what());
+    errno = 0;
+  }
+  return nullptr;
+}
+
+int
+xrtXclbinFreeHandle(xrtXclbinHandle handle)
+{
+  try {
+    free_xclbin(handle);
+    return 0;
+  }
+  catch (const xrt_core::error& ex) {
+    xrt_core::send_exception_message(ex.what());
+    return ex.get();
+  }
+  catch (const std::exception& ex) {
+    send_exception_message(ex.what());
+    return -1;
+  }
+}
+
+#if 0
+// TODO
+// Need a new interface for the C API
+int
+xrtXclbinGetCUNames(xrtXclbinHandle handle, char** names, int* numNames)
+{
+  try {
+    auto xclbin = get_xclbin(handle);
+    const std::vector<std::string> cuNames = xclbin->get_cu_names();
+    // populate numNames if memory is allocated
+    if (numNames)
+      *numNames = cuNames.size();
+    // populate names if memory is allocated
+    if (names) {
+      auto index = 0;
+      for (auto&& name: cuNames) {
+        std::strcpy(names[index++], name.c_str());
+      }
+    }
+    return 0;
+  }
+  catch (const xrt_core::error& ex) {
+    xrt_core::send_exception_message(ex.what());
+    // Set errno globally
+    errno = ex.get();
+    return ex.get();
+  }
+  catch (const std::exception& ex) {
+    send_exception_message(ex.what());
+    // Set errno globally
+    errno = 0;
+    return 0;
+  }
+}
+#endif
+
+int
+xrtXclbinGetXSAName(xrtXclbinHandle handle, char* name, int size, int* ret_size)
+{
+  try {
+    auto xclbin = get_xclbin(handle);
+    const std::string& xsaName = xclbin->get_xsa_name();
```

Comment:
xsaname for variable name, no where else have you used camel case and that is good :-)  Only the C-API function name and sometimes argument names use camel case, but not implementation.

Reply:
Also done 👍